package structDefinitions;

message Properties {
	optional double quality = 1;
	optional string label = 2;
	optional int32 groupId = 3;

}

message Point3D {
	required double x = 1;
	required double y = 2;
	required double z = 3;
}


message Vertex {
	required Point3D point = 1;
	optional Properties prop = 3;
}


message Edge {
	required Point3D v1 = 1;
	required Point3D v2 = 2;
	optional Properties prop = 3;
}


message TriangleFace {
	required Point3D v1 = 1;
	required Point3D v2 = 2;
	required Point3D v3 = 3;
	optional Properties prop = 4;
}


message Block {
	required Point3D v1 = 1;
	required Point3D v2 = 2;
	required Point3D v3 = 3;
	required Point3D v4 = 4;
	optional Properties prop = 5;
}


message MessageInfo {
	enum Type {
		DATA = 1;
		RENDER = 2;
		BREAKPOINT = 3;
		ACCEPTED = 4;
		REJECTED = 5;
		ACK = 6;
		CLEAN = 7;
	}
	
	required Type type = 1;
}

message Header{
	required int32 sizeOfData = 1;
	required bool endOfData = 7;
}

message DataPackage {
	
	repeated Vertex vertexes = 2;
	repeated Edge edges = 3;
	repeated TriangleFace faces = 4;
	repeated Block blocks = 5;
	
}

